compiler/quilt/tools/command.py

```diff
@@ -940,6 +940,53 @@ def install(package, hash=None, version=None, tag=None, force=False, meta_only=F
 
     pkgobj.save_contents()
 
+def materialize(package):
+    if isinstance(package, string_types):
+        node, pkgobj, info = _load(package)
+    elif isinstance(package, nodes.Node):
+        node = package
+    else:
+        raise ValueError("Expected a Node or a package name, but got %r" % package)
+
+    store = PackageStore()
+
+    hashes = set()
+
+    stack = [node]
+    while stack:
```

Comment:
This is reimplementing tree traversal (we now have a helper, right?) and is duplicating the fragment download logic from install. Maybe it'll look cleaner after merging your data_transfer changes?

Reply:
The helper is for core nodes (which we don't have until we "build" the package... yeah, I know, unnecessarily confusing.)

As for duplicating: where do you see that? My data_transfer changes _are_ merged.